lib/irb/completion.rb

```diff
@@ -218,6 +209,33 @@ def self.select_message(receiver, message, candidates, sep = ".")
         end
       end
     end
+
+    def self.ignored_modules
+      # We could cache the result, but this is very fast already.
+      # By using this approach, we avoid Module#name calls, which are
+      # relatively slow when there are a lot of anonymous modules defined.
+      require 'set'
+      s = Set.new
+
+      scanner = lambda do |m|
+        next if s.include?(m) # IRB::ExtendCommandBundle::EXCB recurses.
+        s << m
+        m.constants(false).each do |c|
+          value = m.const_get(c)
+          scanner.call(value) if value.is_a?(Module)
+        end
+      end
+
+      %w(IRB SLex RubyLex RubyToken).each do |module_name|
```

Comment:
You can use ``%i(IRB SLex RubyLex RubyToken)`` to create an array of symbols.

Reply:
Good idea, changed.